common/src/lib.rs

```diff
@@ -21,27 +21,67 @@ pub mod wasm;
 
 use std::collections::BTreeMap;
 
-use serde::{Deserialize, Serialize};
-#[cfg(feature = "service")]
-use uuid::Uuid;
-
+use anyhow::bail;
 #[cfg(feature = "service")]
 pub use log::Item as LogItem;
 #[cfg(feature = "service")]
 pub use log::STATE_MESSAGE;
+use serde::{Deserialize, Serialize};
+#[cfg(feature = "service")]
+use uuid::Uuid;
 
 #[cfg(debug_assertions)]
 pub const API_URL_DEFAULT: &str = "http://localhost:8001";
 
 #[cfg(not(debug_assertions))]
 pub const API_URL_DEFAULT: &str = "https://api.shuttle.rs";
 
-pub type ApiKey = String;
 pub type ApiUrl = String;
 pub type Host = String;
 #[cfg(feature = "service")]
 pub type DeploymentId = Uuid;
 
+#[derive(Clone, Serialize, Deserialize, Debug)]
+#[cfg_attr(feature = "persist", derive(sqlx::Type, PartialEq, Hash, Eq))]
+#[cfg_attr(feature = "persist", serde(transparent))]
+#[cfg_attr(feature = "persist", sqlx(transparent))]
```

Comment:
A naming question, haha: why is the feature called `persist`? Maybe because this `ApiKey` will persist in the `.config` dir?

Reply:
Good question! :smile: I struggled a bit with this name, it's persist now because it enables sqlx, pretty much. We might use it elsewhere in common as well in the future. Very open for naming suggestions here :joy:

Reply:
We might also want to not derive debug for this, so it can't be unintentionally logged :thinking:

Reply:
That's a good point, I believe this is what the secrecy crate does as well. Perhaps we should use that here.